e2e/run_tests.js

```diff
@@ -1,13 +1,78 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-/* eslint-disable no-await-in-loop */
+/* eslint-disable no-await-in-loop, no-console */
 
 const cypress = require('cypress');
 const fse = require('fs-extra');
 const {merge} = require('mochawesome-merge');
 const generator = require('mochawesome-report-generator');
 
+const MAX_FAILED_TITLES = 5;
+
+function getAllTests(results) {
+    const tests = [];
+    results.forEach((result) => {
+        result.tests.forEach((test) => tests.push(test));
+
+        if (result.suites.length > 0) {
+            getAllTests(result.suites).forEach((test) => tests.push(test));
+        }
+    });
+
+    return tests;
+}
+
+function generateStatsFieldValue(stats, failedFullTitles) {
+    let statsFieldValue = `
+| Key | Value |
+|:---|:---|
+| Passing Rate | ${stats.passPercent.toFixed(2)}% |
+| Duration | ${(stats.duration / (60 * 1000)).toFixed(2)} mins |
+| Suites | ${stats.suites} |
+| Tests | ${stats.tests} |
+| :white_check_mark: Passed | ${stats.passes} |
+| :x: Failed | ${stats.failures} |
+| :fast_forward: Skipped | ${stats.skipped} |
+`;
+
+    let failedTests;
+    if (failedFullTitles && failedFullTitles.length > 0) {
```

Comment:
Can you please add some comment on what is being done in the subsequent steps.

Reply:
Sure, now added.